keps/78-dynamically-reclaiming-resources/README.md

```diff
@@ -0,0 +1,169 @@
+# KEP-78:  Dynamically reclaim resources of Pods of a Workload
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Pod Completion Accounting](#pod-completion-accounting)
+    - [Example (<code>batch/job</code>)](#example-)
+  - [API](#api)
+- [Implementation](#implementation)
+  - [Workload](#workload)
+    - [API](#api-1)
+    - [<code>pkg/workload</code>](#)
+    - [Controller](#controller)
+  - [Jobframwork](#jobframwork)
+  - [Batch/Job](#batchjob)
+- [Testing Plan](#testing-plan)
+  - [NonRegression](#nonregression)
+  - [Unit Tests](#unit-tests)
+  - [Integration tests](#integration-tests)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+This proposal allows Kueue to reclaim the resources of a successfully completed Pod of a Workload even before the whole Workload completes its execution. Freeing the unused resources of a Workload earlier may allow more pending Workloads to be admitted. This will allow Kueue to utilize the existing resources of the cluster much more efficiently.
+
+In the remaining of this document:
+1. *Job* refers to any kind job supported by Kueue including `batch/job`, `MPIJob`, `RayJob`, etc.
+1. *Pods of a Workload* means the same as *Pods of a Job* and vice-versa.
+
+
+## Motivation
+
+Currently, the assigned to a Job are reclaimed by Kueue only when the whole Job finishes.
+For Jobs with multiple Pods, the resources will be reclaimed only after the last Pod of the Job finishes.
+This is not efficient as the Jobs might have different needs during execution, for instance the needs of a `batch/job` having the `parallelism` equal to `completions` will decrease with every Pod finishing it's execution.
+
+### Goals
+
+- Utilize the unused resources of the successfully completed Pods of a running Job.
+
+### Non-Goals
+
+- Preempting a Workload or Pods of a Workload to free resources for a pending Workload.
+- Partially admitting a Workload.
+- Monitor the Job's pod execution.
+
```

Comment:
> MPIJob doesn't have partial termination semantics. Probably Ray doesn't either.

Yes, I agree.
I asked about that because I couldn't find mention of other framework jobs.